packages/gatsby-plugin-image/src/components/compat.browser.tsx

```diff
@@ -89,18 +88,19 @@ export function _createCompatLayer(
       const placeholder = fixed.tracedSVG || fixed.base64
 
       if (placeholder) {
-        rewiredProps.placeholder = {
+        image.placeholder = {
           fallback: placeholder,
         }
       }
 
       if (fixed.srcSetWebp) {
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        rewiredProps.images!.sources!.push({
+        image.images.sources!.push({
           srcSet: fixed.srcSetWebp,
           type: `image/webp`,
         })
       }
+      rewiredProps.image = image
     }
 
     if (fluid) {
```

Comment:
I think the property is exclusive, right? Either fixed or fluid but not both and not neither? The branching is not exclusive but both branches set the `.image` property while the type requires the `image` property.